scripts/tests/twister/test_environment.py

```diff
@@ -0,0 +1,579 @@
+
+    assert options.enable_size_report
+
+    assert options.enable_coverage
+
+    assert options.coverage_platform == ['dummy_platform']
+
+    assert options.extra_test_args == ['dummy_extra_1', 'dummy_extra_2']
+
+
+TESTDATA_3 = [
+# Causes an error because some option member accesses
+# do not check if it is not None
+#    (
+#        None,
+#        mock.Mock(
+#            generator_cmd='make',
+#            generator='Unix Makefiles',
+#            test_roots=None,
+#            board_roots=None,
+#            outdir=None,
+#        )
+#    ),
+    (
+        mock.Mock(
+            ninja=True,
+            board_root=['dummy1', 'dummy2'],
+            testsuite_root=[
+                os.path.join('dummy', 'path', "tests"),
+                os.path.join('dummy', 'path', "samples")
+            ]
+        ),
+        mock.Mock(
+            generator_cmd='ninja',
+            generator='Ninja',
+            test_roots=[
+                os.path.join('dummy', 'path', "tests"),
+                os.path.join('dummy', 'path', "samples")
+            ],
+            board_roots=['dummy1', 'dummy2'],
+            outdir='dummy_abspath',
+        )
+    ),
+    (
+        mock.Mock(
+            ninja=False,
+            board_root='dummy0',
+            testsuite_root=[
+                os.path.join('dummy', 'path', "tests"),
+                os.path.join('dummy', 'path', "samples")
+            ]
+        ),
+        mock.Mock(
+            generator_cmd='make',
+            generator='Unix Makefiles',
+            test_roots=[
+                os.path.join('dummy', 'path', "tests"),
+                os.path.join('dummy', 'path', "samples")
+            ],
+            board_roots=['dummy0'],
+            outdir='dummy_abspath',
+        )
+    ),
+]
+
+
+@pytest.mark.parametrize(
+    'options, expected_env',
+    TESTDATA_3,
+    ids=[
+#        'no options',
+        'ninja',
+        'make'
+    ]
+)
+def test_twisterenv_init(options, expected_env):
+    with mock.patch(
+        'os.path.abspath',
+        mock.Mock(return_value='dummy_abspath')
+    ):
+        twister_env = twisterlib.environment.TwisterEnv(options=options)
+
+    assert twister_env.generator_cmd == expected_env.generator_cmd
+    assert twister_env.generator == expected_env.generator
+
+    assert twister_env.test_roots == expected_env.test_roots
+
+    assert twister_env.board_roots == expected_env.board_roots
+    assert twister_env.outdir == expected_env.outdir
+
+
+def test_twisterenv_discover():
+    options = mock.Mock(
+        ninja=True
+    )
+
+    with mock.patch(
+        'os.path.abspath',
+        mock.Mock(return_value='dummy_abspath')
+    ):
+        twister_env = twisterlib.environment.TwisterEnv(options=options)
+
+    mock_datetime = mock.Mock(
+        now=mock.Mock(
+            return_value=mock.Mock(
+                isoformat=mock.Mock(return_value='dummy_time')
+            )
+        )
+    )
+
+    with mock.patch.object(
```

Comment:
```suggestion
    with mock.patch.object(
            twisterlib.environment.TwisterEnv,
            'check_zephyr_version',
            mock.Mock()) as mock_czv, \
         mock.patch.object(
            twisterlib.environment.TwisterEnv,
            'get_toolchain',
            mock.Mock()) as mock_gt, \
         mock.patch('twisterlib.environment.datetime', mock_datetime):
```
indentation only

Reply:
I've tried to keep to 1TBS, as it is pretty clear in most cases, but I see that this indentation style is very good for the context manager.